File: 1-two-sum/two-sum.py
class Solution:
  def twoSum(self, nums, target):
        # Create an empty hashmap (dict) to store number -> index
        hashmap = {}
        
        # Loop through the list
        for i, num in enumerate(nums):
            # Find the number needed to reach target
            complement = target - num
            
            # If we have seen the complement before, return the indices
            if complement in hashmap:
                return [hashmap[complement], i]
            
            # Otherwise, store this number and its index
            hashmap[num] = i

File: 1-two-sum/two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Create a HashMap to store numbers and their indices
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            // Check if complement already exists in the map
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            
            // If not, store the current number and its index
            map.put(nums[i], i);
        }
        
        // If no pair found
        return new int[] {};
    }
}
